Ilya_Torch_itorch2001@gmail.com

IlyaTorch opened the pull request:
Iteration 1
    Options --version, --json, --limit, --verbose are ready. Add option --verbose. Split classes into files. Add description of project to file README.md.

IlyaTorch committed:
Iteration 1
Options --version, --json, --limit are ready. Split classes into files.

 Iteration 1
    Options --version, --json, --limit are ready. Split classes into files.

IlyaTorch committed:
Iteration 1
Add option cl  verbose

IlyaTorch committed:
Iteration 1
Add description of project to file README.md

HenadziStantchik commented on final_task/README.md:
```diff
@@ -1,3 +1,61 @@
+
+With the argument `--verbose` program prints all logs in stdout.
+
+Withe the argument `--version` program prints in stdout it's current version and complete it's work.
```
You are supposed to write documentation yourself. Not to copy it from the task.

IlyaTorch replied:
Why? I did parser completely according to the task and all the options are listed in the task.

HenadziStantchik replied:
But if you had inserted example of your program working you could have seen that there are some problems with encoding at least. We do not need to read about which arguments your application take because we know it anyway.  

We want you to describe how does your application work in the documentation and probably to list some peculiarities of your implementation, some explanations and etc.

IlyaTorch replied:
Got it

HenadziStantchik commented on final_task/rss_reader/Entry.py:
```diff
@@ -0,0 +1,77 @@
+        self.article_link = article_link
+        self.summary = summary
+        self.links = links
+        self.parse_html()
```
It is better to make this method to return parsed summary and write it like this:  
`self.summary = self.parse_html(summary)`  

Right now you assign `self.summary = summary` and after that modify it anyway. This assignment is a bit redundant

IlyaTorch replied:
fixed

HenadziStantchik commented on final_task/rss_reader/Entry.py:
```diff
@@ -0,0 +1,77 @@
+        logging.info("function \"print_title\" started")
+
+        print("Title: ", self.title)
+
```
Here and below: you do not really need empty lines here. Code will still be pretty readable without them

IlyaTorch replied:
fixed

HenadziStantchik commented on final_task/rss_reader/Entry.py:
```diff
@@ -0,0 +1,77 @@
+
+        print("Date: ", self.date)
+
+        logging.info("function \"print_date\" finished")
```
Here, above and below. You can make these lines better (logging part). Think about actions you perform here:

1. You perform some logic (write logs) before performing function itself.
2. You perform function.
3. You perform some logic (write logs) after function.

We have a potential for a good decorator here. It is perfect example of decorator usage.

IlyaTorch replied:
fixed

HenadziStantchik commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,90 @@
+            entry.print_summary()
+            entry.print_links()
+            if num == self.limit - 1:
+                break
```
A question:  

Why do you need this here if you already have almost the same check a bit above?

IlyaTorch replied:
I think that I need this condition because in the cicle: 
"for num, entry in enumerate(self.gen_entries()):
a new entry will be created and only after it I'll get False in this condition:
if num == self.limit:
    break 
i.e an additional object will be created.

HenadziStantchik replied:
That is true, but why do you need `if` condition at the top then? you will never hit it before the lower one anyway

IlyaTorch replied:
If limit is equal for example 0 we'll leave the cycle and won't print information about an entry. Or limit is equal to 1 then we'll get the information about entry and leave the cycle because this condition:
 if num == self.limit - 1:
                break
will return true. If this condition would not exist, we would get one more object entry because it creates in the cicle:
"for num, entry in enumerate(self.gen_entries())
These 2 conditions are necessary to not to get one more extra object.
And I don't know what is more important in this case: not to get an extra object or more readable code.

HenadziStantchik replied:
Now I get it. Then it is better to remove upper check and make it so that `--limit` must be > 0 if it was specified by user.

Upper check will trigger only in case if `limit == 0` and in any other case the cycle will break at the bottom check.

HenadziStantchik commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,38 @@
+    arg_parser.add_argument("--verbose", action="store_true", help="Outputs verbose status messages")
+    arg_parser.add_argument("--limit", type=int, default=1, help="Limit news topics if this parameter provided")
+
+    arg_parser_args = arg_parser.parse_args()
```
It is better to move this to a separate function

IlyaTorch replied:
fixed

HenadziStantchik commented:
Please do NOT close this and open new pull request anymore, Our comments do not save and it is hard to track your progress

HenadziStantchik commented:
Here is some feedback about your application work:
1. If you to check version without specifying RSS url, you wil get:
`rss_reader.py: error: the following arguments are required: source`  
It should be that you do not need any RSS url for checking app version.
2. You got some problems with encoding. Check image below:

![image](https://user-images.githubusercontent.com/43743521/68238394-264dbd00-001a-11ea-937d-6ff189bab61d.png)

3. If limit is not specified you will get only 1 piece of news right now, while it should give you the whole available feed
4. Sometimes you do not get image link in "Links" (see the image below)

![image](https://user-images.githubusercontent.com/43743521/68238411-2f3e8e80-001a-11ea-9f96-d1c1e1292aa5.png)

P.S. I used --limit 999 option to test last threee points.

IlyaTorch commented:
1), 3) -fixed
2) Is it necessary?
4) Img tag exists on this page and it has an attribute "alt" while it doesn't have an attribute "src". What do I need to do in this case?

IlyaTorch committed:
Iteration 1
Remove redundants strings and correct method parse_html in the file Entry.py, Remove redundants empty lines in all the files. Make a decorator for logging and move to other file Logging. Combine parsing command line args into function and move it to other file Console_parse. Fixe the error when check version without RSS url. And fixe output: print whole available feed instead of an article.

HenadziStantchik commented:
> 2) Is it necessary?

Yes

> 4) Img tag exists on this page and it has an attribute "alt" while it doesn't have an attribute "src". What do I need to do in this case?

That is certainly strange. Did you visit the url? if there is no src then there should be no image available on the page.

P.S. In the future use labels. Check slack for more info.

HenadziStantchik commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,73 @@
+        self.parsed = feedparser.parse(self.source)
+        logging.info("Handler object created")
+
+    """options of command line:"""
```
Usually docstrings are ituated only at the beginning of the function/class/module. Here specifically it is better to make that a comment

IlyaTorch replied:
fixed

IlyaTorch replied:
And I have a question about the point 4 above:
\<img src="" width="130" height="86" alt="See Photos of the SpeedKore Dodge Charger" align="left" title="See Photos of the SpeedKore Dodge Charger" border="0" \>
this is a string from url "https://news.yahoo.com/rss/"
And this image isn't on the html version of this page. What can I do in this case?

HenadziStantchik replied:
Well, that is still strange, but in this case, when you have no text and no image it seems logical to me to ignore this piece of news in the feed

IlyaTorch replied:
In such articles text exists but img tags has no attribute src

HenadziStantchik commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,28 @@
+import argparse
+from Handler import Handler
+from Logging import logging_decorator
+from Console_parse import get_arguments_from_console
```
I do not see these files in this PR. Are they 3rd-party libraries? If so, please update requirements.txt file

IlyaTorch replied:
I forgot to commit these files.

HenadziStantchik replied:
Still do not see these files. There is nothing for me to review

IlyaTorch committed:
Iteration 1
Remove unsed improts in files Entry.py, rss_reader.py

IlyaTorch commented:
is:open label:question

HenadziStantchik commented:
Also please do not `request review` using site interface, because it triggers a bit different functionality and compells us to create feedbacks abouth your PR and either merge it or close it. Cancel review request please

IlyaTorch committed:
Iteration 1
Edit output of links in file Handler.py; Edit fiel README.md; Add files ConsoleParse.py and Logging.py; Add parsing special symbols and ASCII codes to function parse_html (file Entry.py)

IlyaTorch commented:
When can I cancel review request?

IlyaTorch committed:
Iteration 1
Remove condition extra condition from file Handler.py and add argument limit check; Remove adding img-links without attr src

HenadziStantchik commented:
When launching your application in this way:
`python rss_reader.py https://news.yahoo,com/rss/`
I get an exception:
![image](https://user-images.githubusercontent.com/43743521/68383585-feb83b00-0166-11ea-90d9-78b370e75c91.png)
This bug is inconsistent because this appears not every application run.

HenadziStantchik commented on final_task/rss_reader/Entry.py:
```diff
@@ -0,0 +1,77 @@
+        # decode ASCII codes:
+        while summary.count("&#"):
+            code = summary[summary.find("&#") + 2: summary.find(";")]
+            summary = summary[:summary.find("&#")] + chr(int(code)) + summary[summary.find("&#") + 3 + len(code):]
```
Try to look into some string standart functions like encode(), decode() for fixing encoding. It should be possible to use them to solve encoding problem in a more fast and elegant manner

IlyaTorch replied:
I couldn't find a way how to do it using encode() or decode() functions so what I used function html.unescape() for it.

IlyaTorch committed:
Simplify the method parse_html

IlyaTorch commented:
It's strange because when I launch application using: 
`python rss_reader.py https://news.yahoo,com/rss/`
command I get only my expect-exception.
![image](https://user-images.githubusercontent.com/45897493/68423598-67c5a000-01b3-11ea-887a-fb84a4a38aec.png)

HenadziStantchik commented:
Looks good so far. Keep it up

IlyaTorch committed:
Iteration 1
Add content of setup.py and add fiel __init__ in rss-reader directory

IlyaTorch commented:
I have a problem with wrapping package:
`python setup.py sdist`
`cd dist`
`pip install rss-reader-1.0.tar.gz`
and when I try to run it:
`rss-reader -h`
I get the following error:
`ModuleNotFoundError: No module named 'Handler'`
But when I run the project from directory where it was installed I don't get errors
What could be the problem?
![image](https://user-images.githubusercontent.com/45897493/68479627-d05c5d80-0243-11ea-8fcd-10efae183018.png)

HenadziStantchik commented:
> I have a problem with wrapping package:
> `python setup.py sdist`
> `cd dist`
> `pip install rss-reader-1.0.tar.gz`
> and when I try to run it:
> `rss-reader -h`
> I get the following error:
> `ModuleNotFoundError: No module named 'Handler'`
> But when I run the project from directory where it was installed I don't get errors
> What could be the problem?

This means either you wrapped it not exatly in correct way or the problem is that package name and module name are the same (rss_reader). Try to use sys.path.append to bypass this

IlyaTorch commented:
Where can I use sys.path.append?

IlyaTorch committed:
Iteration 2
Solve problem with wrapping package. Add opportunity wrapping into distribution package

IlyaTorch committed:
Iteration 2
Add folders dist and rss_reader.egg-info to have oppourtunity install CLI utility rss-reader from them

IlyaTorch committed:
Iteration 2
Add instruction CLI utility rss-reader

IlyaTorch committed:
Iteration 2
Add tests for classes Entry and Handler

HenadziStantchik commented on final_task/README.md:
````diff
@@ -1,3 +1,60 @@
+Links:
+[0]  https://news.yahoo.com/graham-trump-ukraine-incoherent-quid-pro-quo-192210175.html (link)
+[1]  http://l2.yimg.com/uu/api/res/1.2/aWhGys7_IW5qIjKaiJpPfg--/YXBwaWQ9eXRhY2h5b247aD04Njt3PTEzMDs-/https://media-mbst-pub-ue1.s3.amazonaws.com/creatr-uploaded-images/2019-11/5527ffe0-00ca-11ea-9f7d-d1e736c1315d (image)
+```
````
Please add instructions on how to launch your app using packaging tools on a clear machine here.

HenadziStantchik commented:
Looks good so far, proceed to the next iteration. Keep in mind that `--limit` should work with `--date`

IlyaTorch committed:
Iteration 2
Modify README.md: add instructions how to set up rss-reader using packaging tools on a clear machine

IlyaTorch committed:
Iteration 2
Change test files for correct run from command line

IlyaTorch committed:
Iteration 2
Add tests in package rss-reader

dzhigailo commented on final_task/rss_reader/Entry.py:
```diff
@@ -0,0 +1,67 @@
+class Entry:
+    """class for every article from http:link...link.rss"""
+    @logging_decorator
+    def __init__(self, title: str = "", date: str = "", article_link: str = "", summary: str = "", links: list = []):
```
During our course we have explained why is it bad to set mutable objects as default arguments. Please, fix it

IlyaTorch replied:
fixed

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,69 @@
+    def option_json(self) -> None:
+        for num, entry in enumerate(self.gen_entries()):
+            self.print_to_json(self.convert_to_dict(entry))
+            if num == self.limit - 1:
```
Insted of checking the limit on each iteration, you can define the number of iteration to be no more than a limit. For example:
```
for item in item_list[:limit]:
 ...
```
In this case you won't need `break` and `if`

dzhigailo replied:
Same above

IlyaTorch replied:
In my case method gen_entries() is a generator and I can't use the following construction:
`
for item in item_list[:limit]:
`
Can I use the following construction:
`
for entry in islice(self.gen_entries(), 0, self.limit):
`
instead of your example?

dzhigailo replied:
I'm not sure if you really need generator in here:)

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,69 @@
+            entry.print_link()
+            entry.print_summary()
+            entry.print_links()
+            if num == self.limit - 1:
```
Same as above

dzhigailo commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,33 @@
+
+    if arg_parser_args.verbose:
+        logging.basicConfig(level=logging.INFO)
+    if arg_parser_args.limit <= 0:
```
If limit<=0, you should raise exception, because it's a malformed input argument

IlyaTorch replied:
fixed

IlyaTorch committed:
Iteration 2
Add raising exception when arg limit <= 0. Replace mutable object list as default arguments in class Entry into tuple and change some functions that depend on it

IlyaTorch committed:
Iteration 3
Add information about news caching to file README.md; Add console argument --date to file ConsoleParse.py; Change structure of class Entry and format json a little to more convenient using option date; Add modules write_json_to_cache, option_date, print_entry, get_entry_from_dict in class Handler to work with --date argument of command line; Add options to work with --date option in file rss_reader.py; Add file cache.json with stored news

HenadziStantchik commented on final_task/rss_reader/cache.json:
```diff
@@ -0,0 +1,122 @@
+      "http://l2.yimg.com/uu/api/res/1.2/7LKu1VqFsBWR.ZGGf.U.zQ--/YXBwaWQ9eXRhY2h5b247aD04Njt3PTEzMDs-/https://s.yimg.com/os/creatr-images/2019-11/fae1bf00-0581-11ea-93ff-f43ed8c16284"
+    ]
+  }
+]
```
Cache files should not be checked in

HenadziStantchik commented:
tar.gz files should not be checked in to GIT

HenadziStantchik commented on final_task/rss_reader/ConsoleParse.py:
```diff
@@ -0,0 +1,14 @@
+
+def get_arguments_from_console():
+    arg_parser = argparse.ArgumentParser(description="Pure Python command-line RSS reader.")
+    group = arg_parser.add_mutually_exclusive_group()
```
(Optional) It is better to make that if the source is specified, then application looks through local cache and prints only news from the specified source

HenadziStantchik commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,111 @@
+import json
+import logging
+from itertools import islice
+from json import JSONDecodeError
```
you have already imported `json` lib above. you can access it using `json.JSONDecodeError`

HenadziStantchik commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,111 @@
+
+    @logging_decorator
+    # read entries from cache.json into list daily_news that have date that is equal to --date DATE
+    # and then raise an exception or print to console
```
Better move this to docstring

HenadziStantchik commented:
Did not discover any issues with Iteration 3 work, good job. Resolve small issues in comments and proceed to Iteration 4

IlyaTorch committed:
Iteration 3
Remove an extra import json in the file Handler.py, Stop checking  cache-files in git

IlyaTorch committed:
Iteration 4
Edit file README.md because folders dist, rss_reader.egg-info arent checked by git. Add opption --to-html: console argument --to-html in ConsoleParse.py, methods option_html and write_to_html in Handler.py

IlyaTorch committed:
Iteration 4
Add possibility of work option --to-html with option --date

IlyaTorch committed:
Iteration 4
Add handling some new exceptions . Add some tests

HenadziStantchik commented on final_task/README.md:
````diff
@@ -1,3 +1,122 @@
+```
+$ python rss_reader.py --date 20191113 --json --verbose
+```
+News is stored in local file cache.json as list of json objects
````
Please add info about iteration 4

HenadziStantchik commented on final_task/rss_reader/RSSReaderException.py:
```diff
@@ -0,0 +1,3 @@
+class RSSReaderException(Exception):
+    def __init__(self, message):
+        self.message = message
```
You do not really need this logic here. With `pass` it will do the same

HenadziStantchik commented:
When there is no `cache.json` file, `FileNotFoundError` is thrown to the stdout with traceback.

Please keep in mind that you need to convert news to at least 2 different formats.

IlyaTorch committed:
Iteration 4
Add info about iteration 4 to file README.md; Simplify RSSReaderException class; Correct mistake when cache-file doesn't exists

IlyaTorch committed:
Iteration 4
Add description of 4 iteration to file README.md; Add pdf output that works with internen-connection and without it; Change output to html: images are taken from cache

IlyaTorch committed:
Iteration 4
Add description of 4 iteration to file README.md; Add pdf output that works with internen-connection and without it; Change output to html: images are taken from cache

Iteration 4
Add file with fonts

Iteration 4
Fix problems with converting to json

Add description of 4 iteration to file README.md; Add pdf output that works with internen-connection and without it; Change output to html: images are taken from cache; Add file with fonts; Fix problems with converting to json

IlyaTorch committed:
Iteration 4
Add file with fonts

IlyaTorch committed:
Iteration 4
Fix problems with converting to json

IlyaTorch committed:
Merge branch 'master' of https://github.com/IlyaTorch/FinalTaskRssParser

HenadziStantchik commented:
1.Conversion does not work with `--date`

If you are not going to implement search in cache by date and by rss url, then it is better to make that if `--date` is specified, RSS url is just ignored

![torch](https://user-images.githubusercontent.com/43743521/69428031-58ad2900-0d41-11ea-834a-eb3dc1875d1b.png)

2. Even it there is internet connection, there are no images in the converted html file
Also there is encoding problem there:
![torch](https://user-images.githubusercontent.com/43743521/69428217-bc375680-0d41-11ea-8d0a-8f8fb00c6d5f.png)

HenadziStantchik commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,272 @@
+        if path[-1] == '\\':
+            path = path[:-1]
+        if path[-1] != '/':
+            path += '/'
```
It is better to use `os.path` lib for any operations with filesystem paths

HenadziStantchik commented:
It is better to have more informative names for converted files.

HenadziStantchik commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,272 @@
+        if path[-1] == '\\':
+            path = path[:-1]
+        if path[-1] != '/':
+            path += '/'
```
Same: `os.path`

IlyaTorch committed:
Iteration 4
Edit checking of path in --to-html and --to-pdf attributes with using os.path, option --date doesn't depend on url attribute

IlyaTorch committed:
Iteration 4
Edit converting t0 html function.

IlyaTorch committed:
Iteration 4
Add requirements of app

IlyaTorch commented:
I have no problems with encoding in all my browsers. I rewrote write_entries_to_html methon but I'm not sure that it'll help. What can I do with it?

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+import urllib.request
+
+import dominate
+from dominate.tags import *
```
Please, do not use `import *`

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+    def write_cache(self, entry_dict: dict):
+        """write new news to cache in json format"""
+        try:
+            entries = json.load(open("cache.json"))
```
Please, use context manager for working with files. You can check if file exists with `os.path.exists()`

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+        if not entries:
+            entries = self.entries
+
+        if type(entries[0]) == dict:
```
Better use `isinstance()` function

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+        pdf.add_page()
+
+        # in case of reading news from cache, entries is the list of dicts and they are converted to Entry-object
+        if type(entries[0]) == dict:
```
Same as above

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+            pdf.write(10, entry.title + '\n\n')
+            pdf.set_font_size(14)
+            pdf.write(10, f"Feed: {entry.feed}\n")
+            pdf.set_font_size(14)
```
Why do you duplicate line 245? Haven't you already set font size 14?

dzhigailo commented on final_task/rss_reader/rss_reader.py:
```diff
@@ -0,0 +1,57 @@
+        handler.option_version()
+    elif arg_parser_args.date:
+        handler.option_date(str(arg_parser_args.date), arg_parser_args.json,
+                            arg_parser_args.to_html, arg_parser_args.to_html,
```
Why do you pass same arguments twice?

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+        """add entries from cache.json into daily_news: list if they have date that is equal to user's --date DATE
+            and then raise an exception or print to console or to outputs to html"""
+        try:
+            entries = json.load(open("cache.json"))
```
Same as above

dzhigailo commented on final_task/rss_reader/Handler.py:
```diff
@@ -0,0 +1,260 @@
+        if not daily_news:
+            raise RSSReaderException("Error. News aren't found")
+        # different cases of command line arguments
+        elif do_pdf and do_html:
```
You can optimize this statements. Instead of elif, you could just check your arguments one by one with if and do some logic, instead of duplicating your code.

HenadziStantchik commented:
> I have no problems with encoding in all my browsers. I rewrote write_entries_to_html methon but I'm not sure that it'll help. What can I do with it?

Then this means that the problem on my side. It is ok then.

HenadziStantchik commented:
I did not discover any more issues with iteration 4

IlyaTorch committed:
Iteration 4
Add tests fo class Entry; Edit output methods and correct writing of date for Entry objects; Edit some Handler class methods using context managers for working with files and checking if file exists with os.path.exists() and using isinstance() function istead of type(*) == ?; Simplify if-else structure in rss_reader.py file

IlyaTorch committed:
Iteration 4
Edit converting to html and pdf functions for case of several images on the page

IlyaTorch committed:
Iteration 4
Add some tests

IlyaTorch commented:
Please look at my tests. Can you explain me why, when I add new tests, % of coverage doesn't change?

IlyaTorch commented:
And how do the files for which I didn't write a single test show 100% coverage? I check coverage so
`coverage run rss_reader.py`
`coverage report -m` or  `coverage html`

IlyaTorch commented:
![image](https://user-images.githubusercontent.com/45897493/69484489-d9088280-0e44-11ea-90f8-184f6215ce95.png)
I wrote tests only for files Entry.py and Hanler.py

HenadziStantchik commented:
Coverage for exceptions is 100% because they do not contain any statements that need to be tested, only class declarations.

Coverage for logging is 100% because you decorated almost every function with logging decorator, so it implicitly called in each your test, so the coverage thinks that you tested it.

If you wrote tests only for entry and handler and there is some percentage on other modules, that means that the functions you wrote tests for call some functions from other modules and this makes coverage think that you tested these functions too.

Try to check some documentation on coverage, I am sure you will find solution to your problem

IlyaTorch committed:
Iteration 4
Add more tests for Entry and Handler classes

IlyaTorch committed:
Iteration 4
Add more tests for Entry and Handler classes

IlyaTorch committed:
Merge branch 'master' of https://github.com/IlyaTorch/FinalTaskRssParser

IlyaTorch committed:
Iteration 4

IlyaTorch committed:
Iteration 4
Move tests to the rss_reader folder; Add prints when html and pdf documents were created

IlyaTorch committed:
Iteration 4
Move tests to rss_reader folder

IlyaTorch committed:
Iteration 4
Add dockstrings;

IlyaTorch committed:
Iteration 4
Change memu options; Delete extra exceptions in Handler

IlyaTorch committed:
add fonts

IlyaTorch committed:
add fonts

HenadziStantchik commented on final_task/README.md:
````diff
@@ -1,3 +1,144 @@
+```
+
+
+5 mains files of project:
+* rss_reader.py - the file which runs the application
+* ConsoleParse.py - contains code which parses arguments from console
+* Entry.py - contains class Entry which represent an article
+* Handler.py - contains class Handler which performes functions of processing objects Entry
+* Logging.py - contains decorator for printing loggs in stdout
````
It would be nice if you kept it up to date.

HenadziStantchik commented:
Here is feedback on your application:
Iteration 1:
`--json` does not work with Cyrillic symbols.
Iteration 2:
Installs fine, all iterations work.
Iteration 3:
User should be able specify date in `YYYYMMDD`:
![image](https://user-images.githubusercontent.com/43743521/70058955-f2ea5800-15f0-11ea-9817-0f36bd53c78e.png)
Iteration 4:
Everything works fine.


Commit messages are very informative and correspond to the recommended commit message style.
Code is well structured and readable, but there are some large methods which lower readability a bit. The code corresponds to the PEP8 guidelines.
Tests are decent, but some of them are failing `test_write_entries_to_html` (and without). keep in mind that tests should not depend on any external factors such as OS or etc.  Coverage percentage is high enough.

IlyaTorch committed:
Iteration 4
Add work --json with Cyrillic symbols. Correct date format  in YYYYMMDD

IlyaTorch committed:
Iteration 4
Edit saving of the news cache

IlyaTorch committed:
Iteraton 4
Add search by url in date option

IlyaTorch committed:
Iteration 5
Add colorize option

IlyaTorch committed:
Update README.md

IlyaTorch committed:
Add some changes to structure of classes according to the mechanism of data hiding
